src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqResponsibilitiesRepository.kt

```diff
@@ -185,6 +188,46 @@ class JooqResponsibilitiesRepository(
         }
     }
 
+    override fun getResponsibilitiesWithCommentsForTouchstone(touchstoneVersionId: String): List<ResponsibilitySetWithComments> =
+            this.dsl.select(Tables.RESPONSIBILITY_SET.ID, Tables.MODELLING_GROUP.ID)
+                    .fromJoinPath(Tables.RESPONSIBILITY_SET, Tables.MODELLING_GROUP)
+                    .where(Tables.RESPONSIBILITY_SET.TOUCHSTONE.eq(touchstoneVersionId))
+                    .map {
+                        val responsibilitySetId = it[Tables.RESPONSIBILITY_SET.ID] as Int
+                        val records = dsl
+                                .select(Tables.SCENARIO_DESCRIPTION.ID, Tables.RESPONSIBILITY.ID)
+                                .fromJoinPath(Tables.RESPONSIBILITY_SET, Tables.RESPONSIBILITY, Tables.SCENARIO, Tables.SCENARIO_DESCRIPTION)
+                                .where(Tables.RESPONSIBILITY.RESPONSIBILITY_SET.eq(responsibilitySetId))
+                                .and(Tables.RESPONSIBILITY.IS_OPEN)
+                                .fetch()
+                                .intoMap(Tables.SCENARIO_DESCRIPTION.ID)
+                        val responsibilities = scenarioRepository.getScenarios(records.keys).map { scenario ->
```

Comment:
Is there a danger this get will get scenarios from out of touchstone if just filtering on the scenario description id? Scenario description not touchstone-specific (but scenario is). 

Other than that, I think the logic here looks right! I think my tendency would have been to try to reduce the number of queries by doing one big query to get the responsibilities and then grouped by responsibility set id, then done another query to get the latest comments. But this is probably more readable, and if performance isn't an issue no reason to limit number of queries.

Reply:
> Other than that, I think the logic here looks right! I think my tendency would have been to try to reduce the number of queries by doing one big query to get the responsibilities and then grouped by responsibility set id, then done another query to get the latest comments.

Done - comments on this change in particular very welcome